megamek/src/megamek/client/ui/swing/util/UIUtil.java

```diff
@@ -521,11 +524,88 @@ public static int getScaledScreenHeight(DisplayMode currentMonitor) {
+     * Baseline is 1080p (1920x1080).
+     * 
+     * @param component The component to check
+     * @return The resolution scale factor for the containing monitor
+     */
+    public static double getResolutionScaleFactor(Component component) {
+        return getResolutionScaleFactor(component, REFERENCE_RESOLUTION);
+    }
+    
+    /**
+     * Gets the resolution scale factor for the monitor containing the specified component.
+     * 
+     * @param component The component to check
+     * @param referenceResolution The reference resolution width/height to use for scaling
+     * @return The resolution scale factor for the containing monitor
+     */
+    public static double getResolutionScaleFactor(Component component, Dimension referenceResolution) {
+        final Dimension logicalScreenSize = UIUtil.getScaledScreenSize(component);
+        final double scaleFactorX = logicalScreenSize.width / referenceResolution.getWidth();
+        final double scaleFactorY = logicalScreenSize.height / referenceResolution.getHeight();
+        return Math.max(MINIMUM_RESOLUTION_SCALE_FACTOR, Math.min(scaleFactorX, scaleFactorY));
+    }
+    
+    /**
+     * Calculate the DPI scale factor for a component
+     * 
+     * @param component The component to get scaling information from
+     * @return The scaling factor based on DPI
+     */
+    public static float getDpiScaleFactor(Component component) {
+        GraphicsConfiguration gc = null;
+        if (component != null) {
+            gc = component.getGraphicsConfiguration();
+        }
+        if (gc == null) {
+            // Fallback to default GraphicsEnvironment if component doesn't have a GraphicsConfiguration
+            gc = GraphicsEnvironment.getLocalGraphicsEnvironment().getDefaultScreenDevice().getDefaultConfiguration();
+        }
+        // Get screen resolution
+        int dpi = Toolkit.getDefaultToolkit().getScreenResolution();
+        // Calculate scale factor (compared to reference 96 DPI)
+        return (float) dpi / DEFAULT_DISPLAY_PPI;
+    }
+
     /**
      * @return The height of the screen taking into account display scaling
      */
     public static Dimension getScaledScreenSize(Component component) {
-        return getScaledScreenSize(component.getGraphicsConfiguration().getDevice().getDisplayMode());
+        GraphicsConfiguration gc = null;
+        if (component != null) {
+            gc = component.getGraphicsConfiguration();
+        }
+        if (gc == null) {
+            try {
+                gc = GraphicsEnvironment.getLocalGraphicsEnvironment()
+                    .getDefaultScreenDevice()
+                    .getDefaultConfiguration();
+            } catch (HeadlessException e) {
+                return new Dimension(800, 600); 
+            }
+        }
+        Rectangle bounds = gc.getBounds();
+        return new Dimension(bounds.width, bounds.height);
```

Comment:
Although this is short enough to not need it, please have more descriptive variable names in the future.

Also, is there a reason we aren't logging the exception? I know it's being dealt with, don't need to pass the exception in just a comment about it would be a good thing in the logs.